Lesson-7/FileManager/FileManager/Command.cs

```diff
@@ -0,0 +1,465 @@
+
+                    Console.ReadKey();
+
+                    Console.Clear();
+
+                    base.SetColorsDefault();
+
+                    return;
+                }
+                if (answer.Contains("n".ToLower()))
+                {
+                    Console.ForegroundColor = ConsoleColor.Red;
+
+                    ShowMessage("Directory not deleted");
+
+                    Console.ReadKey();
+
+                    Console.Clear();
+
+                    base.SetColorsDefault();
+
+                    return;
+                }
+            }
+            else
+            {
+                ShowMessage("Directory not exists");
+
+                Console.ReadKey();
+            }
+        }
+        /// <summary>
+        /// Метод открывает любой ассоциированный файл
+        /// </summary>
+        /// <param name="path"></param>
+        public void OpenFile(string path)
+        {
+            var purePath = path.Replace("open:", "");
+
+            FileInfo fInfo = new FileInfo(purePath);
+
+            if (fInfo.Exists)
+            {
+                ShowMessage("Open file? [y/n]");
+
+                string answer = Console.ReadLine();
+
+                Console.Clear();
+
+                if (answer.Contains("y".ToLower()))
+                {
+                    //Код для открытия файла
+                    var openFile = new Process();
+
+                    openFile.StartInfo.FileName = @$"{fInfo.FullName}";
+
+                    openFile.StartInfo.UseShellExecute = true;
+
+                    openFile.Start();
+                }
+                if (answer.Contains("n".ToLower()))
+                {
+                    return;
+                }
+            }
+            else
+            {
+                return;
+            }
+        }
+        /// <summary>
+        /// Метод для получения списка команд (в обоих режимах)
+        /// </summary>
+        /// <param name="path"></param>
+
+        public void CreateSystemReport(string path)
+        {
+            var purePath = path.Replace("crsysrep:", "");
+
+            ShowMessage("Create current system report? [y/n]");
+
+            string answer = Console.ReadLine();
+
+            Console.Clear();
+
+            if (answer.Contains("y".ToLower()))
+            {
+                //Lesson-7:
+                string saveReportDir = Directory.GetCurrentDirectory() + "\\SystemReports";
+                string templateDir = Directory.GetCurrentDirectory() + "\\ReportTemplates\\ReportTemplate.docx";
+
+                var systemReport = new SystemReportService(templateDir, saveReportDir);
+
+                var driveInfo = base.CreateSystemDrivesReport();
+
+                var sysInfo = new SystemReport()
+                {
+                    Id = systemReport.SessionId,
+                    ReportDate = DateTime.Now,
+                };
+
+                sysInfo.SystemInfo = new List<LogicalDriveInfo>();

```

Comment:
```suggestion
                sysInfo.SystemInfo = new List<LogicalDriveInfo>(driveInfo.Length);
```
Вы заранее знаете сколько элементов будет в листе - указывайте этот размер, что бы снизить потребление ресурсов на расширение внутреннего массива листа